dataland-document-manager/src/main/kotlin/org/dataland/documentmanager/services/conversion/TextToPdfConverter.kt

```diff
@@ -0,0 +1,55 @@
+package org.dataland.documentmanager.services.conversion
+
+import com.itextpdf.kernel.pdf.PdfDocument
+import com.itextpdf.kernel.pdf.PdfWriter
+import com.itextpdf.layout.Document
+import com.itextpdf.layout.element.Paragraph
+import org.dataland.datalandbackendutils.exceptions.InvalidInputApiException
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import org.springframework.web.multipart.MultipartFile
+import java.io.BufferedReader
+import java.io.ByteArrayOutputStream
+import java.io.InputStreamReader
+
+/**
+ * A converter for txt files to the pdf format
+ */
+@Component
+class TextToPdfConverter : FileConverter(
+    allowedMimeTypesPerFileExtension = mapOf(
+        "txt" to setOf("text/plain"),
+    ),
+    clamAvClient = null,
+) {
+    override val logger: Logger = LoggerFactory.getLogger(javaClass)
+
+    override fun validateFileContent(file: MultipartFile, correlationId: String) {
+        if (file.bytes.decodeToString().isBlank()) {
+            throw InvalidInputApiException(
+                "Provided file is empty.",
+                "Provided file is empty.",
+            )
+        }
+    }
+
+    override fun convert(file: MultipartFile, correlationId: String): ByteArray {
+        logger.info("Converting plain text file to pdf document.(correlation ID: $correlationId)")
+        val outputStream = ByteArrayOutputStream()
+        val pdfDocument = PdfDocument(PdfWriter(outputStream))
+        val document = Document(pdfDocument)
+
+        BufferedReader(InputStreamReader(file.inputStream)).use { reader ->
+            var line: String?
+            while (reader.readLine().also { line = it } != null) {
+                document.add(Paragraph(line))
+            }
+        }
```

Comment:
why do you need the var line? couldn't you just it in a proper way?

Reply:
i mean "it"

Reply:
line gets updated in each iteration. We could use the whole while body as the also block but that would hurt understandability a bit too much.